src/js/files/components/UploadOverlay.js

```diff
@@ -94,15 +94,24 @@ export const UploadOverlay = ({ uploads }) => {
     if (uploads.length) {
         const totalRemainingUploadTime = sum(uploads.map(upload => upload.remaining));
         const overallUploadSpeed = sum(uploads.map(upload => upload.uploadSpeed));
-
         const uploadComponents = map(uploads, upload => <UploadItem key={upload.localId} {...upload} />);
 
+        const totalProgress = sum(uploads.map(upload => upload.progress)) / uploads.length;
```

Comment:
You don't actually need to do math here.

You are trying to find out if all of the uploads have `progress === 100`. This will not be `true` if even one upload has `progress !== 100`. That means there is a good chance we don't have to iterate through the entire array (twice) like you are doing above.

Lodash has a convenient function for this called [`every()`](https://lodash.com/docs/4.17.15#every).